### DISCUSSION TITLE
I can't expand collection

### DISCUSSION THREAD

**Discussion by Graciasjr** [pocketbase/pocketbase#3039] [Q&A] [answered]
Hello @ganigeorgiev 
Please i have probleme when I try to expand my collection.
However I think i have respected api rules and use Pocketbase 0.15.3.
That's my request. 
```
Const liste = await locals.pb.collection('collectionname').getFullList(1,{
 filter:`userid='${user.id}`,
 sort:'-created',
 expand:'otherlist(collectionName)'
}
```
Console.log(liste)
But in response,collection haven't been expanded.

**Comment by ganigeorgiev** [comment]
Without information about your collections structure I don't think I can help you.

Details about `expand` you can find in https://pocketbase.io/docs/working-with-relations/#expanding-relations.

_Side-note: The batch size in the legacy syntax `getFullList(1, { ... })` will send 1 item per request which could be very inefficient. I'd advice removing the explicit `1` batch size (aka. just `geFullList({ ... })`) and also to update your JS SDK to v0.16.0+ and the PocketBase executable to v0.17.0+ so that you can benefit of the `skipTotal` performance boost._

**Comment by Graciasjr** [comment]
> Without information about your collections structure I don't think I can help you.

 In my principal collections i have field named `relationFile` which is of `relation` type.
`relationFile` is related to the `File` collection

> also to update your JS SDK to v0.16.0+ and the PocketBase executable to v0.17.0+ so that you can benefit of the `skipTotal` performance boost.

could not updating the executable be the cause? or is there a way to update the exec without altering the data in my collection?

**Reply by ganigeorgiev** [reply]
> In my principal collections i have field named relationFile which is of relation type.
relationFile is related to the File collection

I'm confused. How this is related to the `collectionname` in the above example? Is the relation single or multiple?

Please understand that I have literally no context about your project, nor what what is your collections configuration and rules. Try to provide as much details as possible with your questions to avoid the needless back-and-forth.


> could not updating the executable be the cause?

No.

**Comment by Graciasjr** [comment]
> Without 

> Side-note: The batch size in the legacy syntax `getFullList(1, { ... })` will send 1 item per request which could be very inefficient. I'd advice removing the explicit `1` batch size (aka. just `geFullList({ ... })`) and also to update your JS SDK to v0.16.0+ and the PocketBase executable to v0.17.0+ so that you can benefit of the `skipTotal` performance boost.

Okay thanks i'll correct this. 
Thanks

**Comment by Graciasjr** [comment]
> I'm confused. How this is related to the `collectionname` in the above example? Is the relation single or multiple?
> 
> Please understand that I have literally no context about your project, nor what what is your collections configuration and rules. Try to provide as much details as possible with your questions to avoid the needless back-and-forth.

Ok i'll give you the general context.
I have a first collection named `root` which has several data fields including `relationFile` which is a `relation(single)` type fiel and which is linked to a second collection named `File`. data diagram given below
```
{ 
"RECORD_ID",
      "collectionId": "jgl6hd6i56qgoq4",
      "collectionName": "mainRoot",
      "userId": "test",
      "categorie": "test",
      "type": "test",
      "name": "test",
      "file": "filename.jpg",
      "relationFile": "RELATION_RECORD_ID"
}
```

I have a second collection named `File`which contains all kinds of data
So, i want to make a single request allowing me to retrieve all fields of the first collection and while expanding the fiels of the `File` collection.

Excuse me🙏🙏, i'm also confused because i still don't really understand the link in the use of a `relation` type field and the `extand` params.

**Reply by ganigeorgiev** [reply, marked as answer]
> So, i want to make a single request allowing me to retrieve all fields of the first collection and while expanding the fiels of the File collection.

Sorry, I'm failing to understand what you are trying to do. Additionally the name of the collection in the json code sample differ from the one in your description (`root` vs `mainRoot`).

If you have a direct relation, then you can expand just using the relation field name.

Indirect relation expansion is used when the targetted relation field doesn't exist in the base collection you are querying.

Couple examples:

```js
// direct file relation expansion from the mainRoot collection
await pb.collection("mainRoot").getFullList({
    expand: "relationFile"
})

// indirect mainRoot relation expansion from the file collection
await pb.collection("file").getFullList({
    expand: "mainRoot(relationFile)"
})
```

I don't know how to explain it better. Please check again the example data structure and expand examples shown in https://pocketbase.io/docs/working-with-relations.

Additionally, double check that your client request satisfy the View API rule of the expanded collection.

**Comment by Graciasjr** [comment]
> Couple examples:
> 
> ```js
> // direct file relation expansion from the mainRoot collection
> await pb.collection("mainRoot").getFullList({
>     expand: "relationFile"
> })
> 
> // indirect mainRoot relation expansion from the file collection
> await pb.collection("file").getFullList({
>     expand: "mainRoot(relationFile)"
> })
> ```

I believe this has helped me.
 I believe I have violated the API rules. I had to expand on the 'relationFile' field and not on my 'File' collection. 
Thanks  @ganigeorgiev for the patience.🙏🙏
